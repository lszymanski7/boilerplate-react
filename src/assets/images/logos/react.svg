<?xml version="1.0" encoding="UTF-8"?>
<svg id="react-logo" width="218px" height="193px" viewBox="0 0 218 193" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <path d="M 177.640625 63.019531 C 175.375 62.238281 173.089844 61.511719 170.785156 60.84375 C 171.171875 59.277344 171.527344 57.710938 171.855469 56.132812 C 177.039062 30.828125 173.648438 10.4375 162.050781 3.71875 C 150.933594 -2.714844 132.761719 4 114.402344 20.046875 C 112.589844 21.628906 110.828125 23.257812 109.101562 24.929688 C 107.949219 23.816406 106.773438 22.722656 105.582031 21.65625 C 86.332031 4.511719 67.035156 -2.707031 55.457031 4.007812 C 44.371094 10.464844 41.078125 29.640625 45.730469 53.621094 C 46.203125 56 46.730469 58.359375 47.304688 60.710938 C 44.589844 61.492188 41.949219 62.320312 39.429688 63.203125 C 16.875 71.113281 2.46875 83.488281 2.46875 96.328125 C 2.46875 109.582031 17.941406 122.886719 41.429688 130.933594 C 43.332031 131.589844 45.253906 132.199219 47.195312 132.757812 C 46.5625 135.28125 46.007812 137.824219 45.527344 140.386719 C 41.058594 163.949219 44.542969 182.667969 55.621094 189.089844 C 67.054688 195.710938 86.246094 188.902344 104.949219 172.492188 C 106.433594 171.203125 107.917969 169.832031 109.398438 168.382812 C 111.261719 170.195312 113.175781 171.957031 115.140625 173.664062 C 133.261719 189.3125 151.140625 195.621094 162.203125 189.199219 C 173.640625 182.550781 177.355469 162.445312 172.527344 137.988281 C 172.15625 136.125 171.734375 134.222656 171.253906 132.273438 C 172.597656 131.875 173.917969 131.464844 175.21875 131.035156 C 199.644531 122.90625 215.53125 109.757812 215.53125 96.328125 C 215.53125 83.429688 200.664062 70.972656 177.640625 63.019531 Z M 120.394531 26.949219 C 136.121094 13.199219 150.828125 7.785156 157.542969 11.65625 C 164.675781 15.785156 167.433594 32.449219 162.953125 54.304688 C 162.660156 55.730469 162.34375 57.148438 161.996094 58.554688 C 152.605469 56.414062 143.089844 54.90625 133.503906 54.035156 C 128.007812 46.109375 121.988281 38.5625 115.484375 31.453125 C 117.078125 29.917969 118.710938 28.414062 120.375 26.949219 Z M 65.371094 107.277344 C 67.300781 111.023438 69.320312 114.714844 71.417969 118.347656 C 73.5625 122.070312 75.789062 125.738281 78.105469 129.351562 C 71.511719 128.636719 64.953125 127.582031 58.46875 126.1875 C 60.355469 120.078125 62.671875 113.722656 65.378906 107.285156 Z M 65.371094 85.785156 C 62.71875 79.476562 60.457031 73.273438 58.609375 67.269531 C 64.671875 65.90625 71.15625 64.789062 77.914062 63.941406 C 75.648438 67.476562 73.472656 71.074219 71.382812 74.734375 C 69.28125 78.367188 67.273438 82.050781 65.359375 85.785156 Z M 70.207031 96.539062 C 73.007812 90.679688 76.035156 84.929688 79.285156 79.292969 C 82.503906 73.648438 85.945312 68.144531 89.601562 62.777344 C 95.9375 62.296875 102.429688 62.03125 109 62.03125 C 115.597656 62.03125 122.097656 62.296875 128.4375 62.777344 C 132.046875 68.144531 135.46875 73.632812 138.699219 79.246094 C 141.949219 84.863281 145.011719 90.585938 147.886719 96.40625 C 145.046875 102.289062 141.996094 108.0625 138.734375 113.730469 C 135.507812 119.363281 132.097656 124.882812 128.507812 130.292969 C 122.191406 130.757812 115.664062 130.984375 109 130.984375 C 102.367188 130.984375 95.917969 130.773438 89.714844 130.367188 C 86.03125 124.972656 82.5625 119.441406 79.300781 113.777344 C 76.050781 108.132812 73.019531 102.382812 70.207031 96.53125 Z M 146.628906 118.308594 C 148.765625 114.566406 150.824219 110.789062 152.800781 106.957031 C 155.503906 113.105469 157.894531 119.390625 159.953125 125.789062 C 153.390625 127.269531 146.757812 128.402344 140.082031 129.1875 C 142.328125 125.597656 144.507812 121.972656 146.617188 118.300781 Z M 152.707031 85.796875 C 150.742188 82.046875 148.703125 78.339844 146.59375 74.667969 C 144.503906 71.058594 142.34375 67.492188 140.105469 63.96875 C 146.894531 64.835938 153.410156 65.980469 159.53125 67.382812 C 157.566406 73.636719 155.289062 79.78125 152.707031 85.804688 Z M 109.09375 38.011719 C 113.527344 42.863281 117.710938 47.9375 121.632812 53.210938 C 113.25 52.8125 104.859375 52.8125 96.457031 53.210938 C 100.597656 47.726562 104.84375 42.628906 109.101562 38.011719 Z M 60.035156 11.914062 C 67.164062 7.777344 82.914062 13.683594 99.53125 28.503906 C 100.597656 29.453125 101.664062 30.441406 102.726562 31.460938 C 96.1875 38.574219 90.121094 46.117188 84.570312 54.035156 C 75.015625 54.894531 65.519531 56.375 56.152344 58.46875 C 55.605469 56.289062 55.121094 54.082031 54.6875 51.863281 L 54.6875 51.882812 C 50.675781 31.324219 53.335938 15.824219 60.035156 11.933594 Z M 49.621094 123.957031 C 47.859375 123.449219 46.113281 122.898438 44.378906 122.308594 C 35.347656 119.457031 26.921875 114.960938 19.511719 109.042969 C 15.226562 106.085938 12.355469 101.476562 11.574219 96.316406 C 11.574219 88.527344 23.132812 78.59375 42.425781 71.847656 C 44.851562 70.992188 47.292969 70.21875 49.761719 69.523438 C 52.621094 78.75 56.085938 87.773438 60.132812 96.539062 C 56.039062 105.433594 52.527344 114.589844 49.628906 123.945312 Z M 98.957031 165.632812 C 91.988281 172.058594 83.902344 177.132812 75.101562 180.617188 C 70.421875 182.871094 65.015625 183.078125 60.171875 181.191406 C 53.457031 177.28125 50.660156 162.265625 54.464844 142.117188 C 54.925781 139.730469 55.441406 137.359375 56.023438 135.007812 C 65.488281 137.046875 75.082031 138.441406 84.738281 139.179688 C 90.332031 147.140625 96.441406 154.722656 103.023438 161.878906 C 101.703125 163.160156 100.347656 164.410156 98.964844 165.625 Z M 109.351562 155.292969 C 105.035156 150.617188 100.738281 145.449219 96.53125 139.882812 C 100.609375 140.046875 104.765625 140.125 109 140.125 C 113.34375 140.125 117.628906 140.035156 121.859375 139.847656 C 117.953125 145.203125 113.78125 150.359375 109.351562 155.292969 Z M 164.628906 168.007812 C 164.230469 173.214844 161.703125 178.023438 157.652344 181.292969 C 150.933594 185.203125 136.578125 180.121094 121.085938 166.746094 C 119.316406 165.21875 117.53125 163.585938 115.734375 161.851562 C 122.195312 154.671875 128.160156 147.0625 133.59375 139.074219 C 143.304688 138.253906 152.945312 136.761719 162.453125 134.609375 C 162.882812 136.371094 163.265625 138.09375 163.601562 139.785156 C 165.660156 149.050781 166.007812 158.621094 164.617188 168.015625 Z M 172.355469 122.355469 C 171.175781 122.75 169.980469 123.117188 168.757812 123.472656 C 165.78125 114.207031 162.179688 105.152344 157.976562 96.371094 C 162.023438 87.714844 165.488281 78.792969 168.351562 69.671875 C 170.519531 70.292969 172.640625 70.972656 174.679688 71.667969 C 194.410156 78.480469 206.433594 88.566406 206.433594 96.328125 C 206.433594 104.605469 193.4375 115.34375 172.34375 122.355469 Z M 172.355469 122.355469" fill="#61DAFB" />

    <path d="M 109 77.214844 C 119.511719 77.214844 128.03125 85.777344 128.027344 96.332031 C 128.023438 106.886719 119.5 115.441406 108.988281 115.433594 C 98.476562 115.425781 89.960938 106.859375 89.972656 96.308594 C 89.988281 85.765625 98.503906 77.226562 109 77.226562" fill="#61DAFB" />
</svg>
