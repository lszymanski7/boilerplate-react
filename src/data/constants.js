const BADGES = {
    github: 'https://img.shields.io/badge/GitHub-555555?style=for-the-badge&logo=github',
    version: 'https://img.shields.io/badge/1.0.0-blue?style=for-the-badge&label=Version&labelColor=555555',
    size: 'https://img.shields.io/github/repo-size/lszymanski7/boilerplate-react?style=for-the-badge&label=Size&labelColor=555555',
    license: 'https://img.shields.io/badge/License-MIT-yellow.svg?style=for-the-badge&label=License&labelColor=555555',
    forks: 'https://img.shields.io/github/forks/lszymanski7/boilerplate-react?style=for-the-badge&color=green&label=Forks&labelColor=555555',
    stars: 'https://img.shields.io/github/stars/lszymanski7/boilerplate-react?style=for-the-badge&color=green&label=Stars&labelColor=555555'
}

const LOGOS = {
    github: require('../assets/images/logos/github.svg'),
    react: {
        size_128x128: require('../assets/images/logos/react-128x128.png'),
        size_256x256: require('../assets/images/logos/react-256x256.png'),
        size_512x512: require('../assets/images/logos/react-512x512.png'),
        svg: require('../assets/images/logos/react.svg')
    }
}

const SHAPES = {
    waves: require('../assets/images/shapes/waves.svg')
}

export { BADGES, LOGOS, SHAPES }
