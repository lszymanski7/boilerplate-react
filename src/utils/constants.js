// React Logo

const logo = {
    size_128x128: require('../assets/images/logos/react-128x128.png'),
    size_256x256: require('../assets/images/logos/react-256x256.png'),
    size_512x512: require('../assets/images/logos/react-512x512.png')
}

export { logo as default }
