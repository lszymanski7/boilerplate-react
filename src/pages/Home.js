import React from 'react'
import { BADGES, LOGOS } from '../data/constants'

const Home = () => {
    // Application name
    const name = 'React Boilerplate'

    return (
        <div className="home">
            <img
                alt="Application Logo"
                className="home__app-logo spin"
                src={LOGOS.react.svg}
            />
            <h1 className="home__app-name">{name}</h1>
            <div className="home__badges">
                <a href="https://github.com/lszymanski7/boilerplate-react">
                    <img
                        alt="GitHub Badge"
                        src={BADGES.github}
                    />
                </a>
                <img
                    alt="Version Badge"
                    src={BADGES.version}
                />
                <img
                    alt="Size Badge"
                    src={BADGES.size}
                />
                <a href="https://github.com/lszymanski7/boilerplate-react/blob/main/LICENSE.md">
                    <img
                        alt="License Badge"
                        src={BADGES.license}
                    />
                </a>
                <img
                    alt="Forks Badge"
                    src={BADGES.forks}
                />
                <img
                    alt="Stars Badge"
                    src={BADGES.stars}
                />
            </div>
        </div>
    )
}

export { Home as default }
