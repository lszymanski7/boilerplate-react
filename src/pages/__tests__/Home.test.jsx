import React from 'react'
import { render, screen } from '@testing-library/react'
import Home from '../Home'
import { BADGES } from '../../data/constants'

describe('Home', () => {
    it('should be rendered correctly.', () => {
        const { container } = render(<Home />)
        expect(container).toMatchSnapshot()
    })

    it('should have the correct application logo.', () => {
        render(<Home />)
        const img = screen.getAllByRole('img')[0]
        expect(img).toHaveAttribute('alt', 'Application Logo')
        expect(img).toHaveAttribute('src', 'react.svg')
    })

    it('should have the correct application name.', () => {
        render(<Home />)
        const name = 'React Boilerplate'
        const h1 = screen.getByRole('heading', { level: 1 })
        expect(h1).toHaveTextContent(name)
    })

    it('should have six badges.', () => {
        render(<Home />)
        const badges = screen.getAllByAltText('Badge', { exact: false })
        expect(badges).toHaveLength(6)
    })

    it('should have badges with the correct src attribute.', () => {
        render(<Home />)
        const badges = screen.getAllByAltText('Badge', { exact: false })
        expect(badges[0]).toHaveAttribute('src', BADGES.github)
        expect(badges[1]).toHaveAttribute('src', BADGES.version)
        expect(badges[2]).toHaveAttribute('src', BADGES.size)
        expect(badges[3]).toHaveAttribute('src', BADGES.license)
        expect(badges[4]).toHaveAttribute('src', BADGES.forks)
        expect(badges[5]).toHaveAttribute('src', BADGES.stars)
    })

    it('should have two links.', () => {
        render(<Home />)
        const links = screen.getAllByRole('link')
        expect(links).toHaveLength(2)
    })

    it('should have links with the correct href attribute.', () => {
        render(<Home />)
        const links = screen.getAllByRole('link')
        expect(links[0]).toHaveAttribute('href', 'https://github.com/lszymanski7/boilerplate-react')
        expect(links[1]).toHaveAttribute('href', 'https://github.com/lszymanski7/boilerplate-react/blob/main/LICENSE.md')
    })
})
