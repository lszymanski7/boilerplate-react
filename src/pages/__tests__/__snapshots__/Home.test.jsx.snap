// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should be rendered correctly. 1`] = `
<div>
  <div
    class="home"
  >
    <img
      alt="Application Logo"
      class="home__app-logo spin"
      src="react.svg"
    />
    <h1
      class="home__app-name"
    >
      React Boilerplate
    </h1>
    <div
      class="home__badges"
    >
      <a
        href="https://github.com/lszymanski7/boilerplate-react"
      >
        <img
          alt="GitHub Badge"
          src="https://img.shields.io/badge/GitHub-555555?style=for-the-badge&logo=github"
        />
      </a>
      <img
        alt="Version Badge"
        src="https://img.shields.io/badge/1.0.0-blue?style=for-the-badge&label=Version&labelColor=555555"
      />
      <img
        alt="Size Badge"
        src="https://img.shields.io/github/repo-size/lszymanski7/boilerplate-react?style=for-the-badge&label=Size&labelColor=555555"
      />
      <a
        href="https://github.com/lszymanski7/boilerplate-react/blob/main/LICENSE.md"
      >
        <img
          alt="License Badge"
          src="https://img.shields.io/badge/License-MIT-yellow.svg?style=for-the-badge&label=License&labelColor=555555"
        />
      </a>
      <img
        alt="Forks Badge"
        src="https://img.shields.io/github/forks/lszymanski7/boilerplate-react?style=for-the-badge&color=green&label=Forks&labelColor=555555"
      />
      <img
        alt="Stars Badge"
        src="https://img.shields.io/github/stars/lszymanski7/boilerplate-react?style=for-the-badge&color=green&label=Stars&labelColor=555555"
      />
    </div>
  </div>
</div>
`;
